### ISSUE TITLE
AR controller PcoketPC

### ISSUE THREAD

**Issue by walksanatora** [SirEndii/AdvancedPeripherals#382]
### Describe your idea

attaching a AR controller to a PocketPC allows you to controll the players AR glasses from within their inventory

### Describe alternatives you've considered if you've any

Having another PC in a loaded chunk conencted to the AR controller to interact with the AR glasses

### Additional context

The ability to use a pocket PC would be nice thatway you dont have to worry about loaded chunks when drawing/getting data from a computer

**Comment by robberer** [comment]
I've tried to implement this myself but failed at https://github.com/SirEndii/AdvancedPeripherals/blob/e78620b85a4ac9180e53735553b8252f9926f9aa/src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/peripheral/ARControllerPeripheral.java .

The AR Controller extends another class than the other pocket-pc upgrades and I didn't found a solution. @SirEndii  can you have a look at my progress. Maybe it is easy to solve.  https://github.com/robberer/AdvancedPeripherals/commit/f86a22b0ec0c75c36cc264b43b75a647428e8f59

**Comment by SirEndii** [comment]
Not sure exactly what you mean, the `ARControllerTile` class and the `ARControllerPeripheral` class extend both the same classes as the other tile/peripheral classes

Your commit looks okay on first sight - but I see that you're doing this on 1.16
I doubt that I'll publish a 1.16 version anytime soon, if it'll even happen at all. I recommend doing that for 1.18.

**Comment by robberer** [comment]
I mean that `ARControllerPeripheral` is not extending `BasePeripheral<IPeripheralOwner>` like all the other pocket-pc equipable peripherals. I don't know how to handle this.

I have to use 16.5 because my client cant run versions higher version due to opengl version not supported.

**Comment by SirEndii** [comment]
I don't want to know how old your pc is that it does not support that opengl version

> I mean that ARControllerPeripheral is not extending BasePeripheral<IPeripheralOwner> like all the other pocket-pc equipable peripherals. I don't know how to handle this.

There shouldn't be any difference, do you get an error if you try it using it anyway?

**Comment by robberer** [comment]
:-) my pc is quite modern but the x200 of my 5 yo coop partner is not.

I'm stuck in the build process.

![Screenshot_2023-04-10_22-29-16](https://user-images.githubusercontent.com/6204883/230968763-a69a572a-e757-469f-b760-0d4bd6d6356e.png)

**Comment by SirEndii** [comment]
Yeah the BasePeriphera uses a block entity owner as the peripheral owner
You could try to change `BasePeripheral<TileEntity...` to `BasePeripheral<IPeripheralOwner>`
